subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+	uint8_t raw_event[HCI_EVENT_PACKET_MAX_SIZE];
+} cmd_complete_or_status;
+
+static bool command_generates_command_complete_event(uint16_t hci_opcode)
+{
+	switch (hci_opcode) {
+	case SDC_HCI_OPCODE_CMD_LC_DISCONNECT:
+	case SDC_HCI_OPCODE_CMD_LE_SET_PHY:
+	case SDC_HCI_OPCODE_CMD_LC_READ_REMOTE_VERSION_INFORMATION:
+	case SDC_HCI_OPCODE_CMD_LE_CREATE_CONN:
+	case SDC_HCI_OPCODE_CMD_LE_CONN_UPDATE:
+	case SDC_HCI_OPCODE_CMD_LE_READ_REMOTE_FEATURES:
+	case SDC_HCI_OPCODE_CMD_LE_ENABLE_ENCRYPTION:
+	case SDC_HCI_OPCODE_CMD_LE_EXT_CREATE_CONN:
+		return false;
+	default:
+		return true;
+	}
+}
+
+/* Return true if the host has been using both legacy and extended HCI commands
+ * since last HCI reset.
+ */
+static bool is_host_using_legacy_and_extended_commands(uint16_t hci_opcode)
+{
+	switch (hci_opcode) {
+#if defined(CONFIG_BT_BROADCASTER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_RESPONSE_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_ENABLE:
+	case SDC_HCI_OPCODE_CMD_LE_READ_MAX_ADV_DATA_LENGTH:
+	case SDC_HCI_OPCODE_CMD_LE_READ_NUMBER_OF_SUPPORTED_ADV_SETS:
+	case SDC_HCI_OPCODE_CMD_LE_REMOVE_ADV_SET:
+	case SDC_HCI_OPCODE_CMD_LE_CLEAR_ADV_SETS:
+#endif  /* CONFIG_BT_BROADCASTER */
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_CENTRAL)
+	case SDC_HCI_OPCODE_CMD_LE_EXT_CREATE_CONN:
+#endif
+		if (type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_NONE) {
+			type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_EXTENDED;
+			return false;
+		}
+		return type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_LEGACY;
+
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_READ_ADV_PHYSICAL_CHANNEL_TX_POWER:
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_RESPONSE_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_CENTRAL)
+	case SDC_HCI_OPCODE_CMD_LE_CREATE_CONN:
+#endif  /* CONFIG_BT_CENTRAL */
+		if (type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_NONE) {
+			type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_LEGACY;
+			return false;
+		}
+		return type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_EXTENDED;
+	case SDC_HCI_OPCODE_CMD_CB_RESET:
+		type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_NONE;
+		break;
+	default:
+		/* Ignore command */
+		break;
+	}
+
+	return false;
+}
+
+static void encode_command_status(uint8_t * const event,
+				  uint16_t hci_opcode,
+				  uint8_t status_code)
+{
+
+	struct bt_hci_evt_hdr *evt_hdr = (void *)event;
```

Comment:
Why cast to void instead of proper type?

Reply:
- [ ] cast to proper type